tests/countries/test_trinidad_and_tobago.py

```diff
@@ -0,0 +1,243 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import OPTIONAL
+from holidays.countries.trinidad_and_tobago import TrinidadAndTobago, TT, TTO
+from tests.common import CommonCountryTests
+
+
+class TestTrinidadAndTobago(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1963, 2050)
+        super().setUpClass(TrinidadAndTobago, years=years)
+        cls.no_estimated_holidays = TrinidadAndTobago(years=years, islamic_show_estimated=False)
+        cls.optional_holidays = TrinidadAndTobago(categories=OPTIONAL, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(TrinidadAndTobago, TT, TTO)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(TrinidadAndTobago(years=1962))
+        self.assertNoHolidays(TrinidadAndTobago(categories=OPTIONAL, years=1962))
+
+    def test_new_years_day(self):
+        self.assertHolidayName("New Year's Day", (f"{year}-01-01" for year in range(1963, 2050)))
```

Comment:
Same goes for other cases (I think the newly added [`test_sierra_leone.py`](https://github.com/vacanza/holidays/blob/0ee2d777ed33a1c3d06181d068eb5564d271c6aa/tests/countries/test_sierra_leone.py#L19)) got some nice examples here:
```suggestion
    def test_new_years_day(self):
        name = "New Year's Day"
        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1963, 2050)))
        dt = (
            "2012-01-02",
            "2017-01-02",
            "2023-01-02",
        )
        self.assertHolidayName(f"{name} (observed)", self.no_estimated_holidays, dt)
        self.assertNoNonObservedHoliday(dt)
```